unittest/yaml_util_test.py

```diff
@@ -279,6 +279,49 @@ def test_bool(self):
                       bindings.get('root'))
      self.assertEqual(bindings.get('root'), bindings.get('copy'))
 
+  def test_update_yml_source(self):
+    yaml = """
+a: A
+b: 0
+c:
+  - A
+  - B
+d:
+  child:
+    grandchild: x
+e:
+"""
+    fd, temp_path = tempfile.mkstemp()
+    os.write(fd, yaml)
```

Comment:
Does this leak temp_path or is the file tied to the process lifetime? The docs arent clear. I would think they persist beyond the test. If so at least os.remove them at the end of this method on success.